connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -94,7 +94,7 @@
     private final Plugins plugins;
     private final ConnectMetrics metrics;
     private final WorkerMetricsGroup workerMetricsGroup;
-    private final ConnectorStatusMetricsGroup connectorStatusMetricsGroup;
+    private ConnectorStatusMetricsGroup connectorStatusMetricsGroup;
```

Comment:
So IIUC, the only reason we can't make this final is because we need a herder to pass to the `ConnectorStatusMetricsGroup` instance. One way to get around that is to avoid passing in the herder and instead pass in a function that does what the `ConnectorStatusMetricsGroup` needs the herder to do: get the task status for a task ID. 

For example, we could define a protected method on `Worker`, such as `taskStatus(ConnectorTaskId)` that returns `ConnectorStateInfo.TaskState`. Then, we could change `ConnectorStatusMetricsGroup` to take a `Function<ConnectorTaskId, TaskState>`, and the other constructor parameters are simple Maps. That means:
1) the class becomes pretty straightforward; and
2) the class is much easier to test, because we don't need a herder